lib/pbench/agent/utils.py

```diff
@@ -188,3 +190,33 @@ def collect_local_info(pbench_bin):
         hostdata[arg] = cp.stdout.strip() if cp.stdout is not None else ""
 
     return (version, seqno, sha1, hostdata)
+
+
+# From https://stackoverflow.com/questions/106179/regular-expression-to-match-dns-hostname-or-ip-address
+_allowed = re.compile(
+    r"(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])"
+)
```

Comment:
I have several suggestions, here:

- Add `flags=re.IGNORECASE` to the `re.compile()` call so that you don't need to include both letter ranges, which makes the expression considerably shorter.
- Use non-capturing groups, which improves efficiency, since we don't actually use the matches.
- RFC 1123 states, "the restriction on the first character is relaxed to allow either a letter or a digit," which has the side effect of allowing your algorithm to match either dotted-decimal quartets or FQDNs.
- So, I think you can render this as:
```
_allowed = re.compile(
    r"[A-Z0-9](?:[A-Z0-9\-]{0,61}[A-Z0-9])?(?:\.[A-Z0-9](?:[A-Z0-9\-]{0,61}[A-Z0-9])?)*",
    flags=re.IGNORECASE
)
```
That is, a case-blind comparison seeking a letter-or-digit, optionally followed by a sequence of zero to 61 letter-or-digit-or-hyphens followed by a letter-or-digit, followed by zero or instances of that same pattern preceded by a dot.

Note that I moved the repetition from the first group to the last: this alleviates backtracking when the RegEx engine doesn't find a match.

Reply:
Slow on the uptake.  I'll modify this.